lib/pengine/tests/native/native_find_rsc_test.c

```diff
@@ -0,0 +1,683 @@
+    assert_null(native_find_rsc(mysql_group_0, "mysql-group", cluster01, pe_find_any|pe_find_current));
+    assert_null(native_find_rsc(mysql_group_0, "mysql-group", cluster01, pe_find_anon|pe_find_current));
+    assert_ptr_equal(mysql_group_1, native_find_rsc(mysql_group_1, "mysql-group", cluster01, pe_find_any|pe_find_current));
+    assert_ptr_equal(mysql_group_1, native_find_rsc(mysql_group_1, "mysql-group", cluster01, pe_find_anon|pe_find_current));
+    assert_null(native_find_rsc(mysql_group_1, "mysql-group", cluster02, pe_find_any|pe_find_current));
+    assert_null(native_find_rsc(mysql_group_1, "mysql-group", cluster02, pe_find_anon|pe_find_current));
+
+    /* Fails because incorrect flags were given along with base name. */
+    assert_null(native_find_rsc(mysql_group_0, "mysql-group", NULL, pe_find_current));
+    assert_null(native_find_rsc(mysql_group_1, "mysql-group", NULL, pe_find_current));
+
+    /* And then we check failure possibilities again, except passing mysql_clone_group
+     * instead of mysql_group_X as the first argument to native_find_rsc.
+     */
+
+    /* Fails because pe_find_current is required if a node is given. */
+    assert_null(native_find_rsc(mysql_clone_group, "mysql-group:0", cluster02, 0));
+    assert_null(native_find_rsc(mysql_clone_group, "mysql-group:1", cluster01, 0));
+
+    /* Check that the resource is running on the node we expect. */
+    assert_ptr_equal(mysql_group_0, native_find_rsc(mysql_clone_group, "mysql-group:0", cluster02, pe_find_current));
+    assert_ptr_equal(mysql_group_0, native_find_rsc(mysql_clone_group, "mysql-group", cluster02, pe_find_any|pe_find_current));
+    assert_ptr_equal(mysql_group_0, native_find_rsc(mysql_clone_group, "mysql-group", cluster02, pe_find_anon|pe_find_current));
+    assert_ptr_equal(mysql_group_1, native_find_rsc(mysql_clone_group, "mysql-group:1", cluster01, pe_find_current));
+    assert_ptr_equal(mysql_group_1, native_find_rsc(mysql_clone_group, "mysql-group", cluster01, pe_find_any|pe_find_current));
+    assert_ptr_equal(mysql_group_1, native_find_rsc(mysql_clone_group, "mysql-group", cluster01, pe_find_anon|pe_find_current));
+}
+
+static void
+clone_group_member_rsc(void **state) {
+    pe_resource_t *mysql_proxy = NULL;
+
+    /* Find the "mysql-proxy" resource, a member of "mysql-group". */
+    for (GList *iter = mysql_clone_group->children; mysql_proxy == NULL && iter != NULL; iter = iter->next) {
+        pe_resource_t *rsc = (pe_resource_t *) iter->data;
+
+        if (strcmp(rsc->id, "mysql-group:0") == 0) {
+            for (GList *iter2 = rsc->children; iter2 != NULL; iter2 = iter2->next) {
+                pe_resource_t *child = (pe_resource_t *) iter2->data;
+
+                if (strcmp(child->id, "mysql-proxy:0") == 0) {
+                    mysql_proxy = child;
+                    break;
+                }
+            }
+        }
```

Comment:
You probably want to break at the end of the if block as well

Reply:
This is handled by the `mysql_proxy == NULL` condition in the for loop, but that's probably too subtle.  I kind of hate adding extra end conditions into the for loop.  I don't know why I did it here.  I'll change it to a break.